### ISSUE TITLE
Suggestion: 'repeat order' function for Amadron tablet

### ISSUE THREAD

**Issue by wrincewind** [TeamPneumatic/pnc-repressurized#202]
It's not a terribly complicated idea, but it could be useful for setting up farms, automation, and such. Basically you can set up a standing order to sell or buy an item. For example, 'sell 14 buckets of petroleum for 14 emeralds', then every time the tank has 14 buckets of petroleum, a drone will float in and take the fluid, then dispense emeralds into the designated container. Similarly you could have 'buy a stack of wheat for 2 emeralds' and each time 2 emeralds are put into the container, a drone comes in and exchanges it for wheat.

**Comment by desht** [comment]
Hmm, this might be achievable by marking an order as "recurring" in the tablet interface.  Then the tablet would periodically check its inventory and/or tank for the resources and auto-place an order if there's sufficient.

This would almost certainly require the player to keep the tablet in their inventory, but I think that's a reasonable trade-off.

**Comment by wrincewind** [comment]
"This would almost certainly require the player to keep the tablet in their inventory, but I think that's a reasonable trade-off."

A possibility that springs to mind is maybe making a new colour of logistic frame to handle the logic? I don't know what the code involves for those, but binding the frames to an Amadron tablet or even to a specific order (select order as repeating, select input container, select output container) sounds doable and would negate that requirement. It would complicate things a tad though, so maybe its not worth the payoff. Food for thought, right?

**Comment by desht** [comment]
An Amadron Frame might be useful, yeah... place it on an inventory or tank, configure with the logistics configurator, which would pop up an Amadron-like interface.  Select a resource, and whenever your inventory/tank has enough of that, an order would be auto-placed.

Could work...

**Comment by KirinDave** [comment]
+1 on the idea of repeat interactions with Amadron. This is such a cool and flexible idea it could literally be the backbone of a skyblock pack.

**Comment by desht** [comment]
Yeah, I need to pull this one off the back burner.  I have been mulling over a few ideas, but haven't come up with a satisfactory solution yet (the Amadron Frame idea isn't really workable, sadly).

The best bet may just be to mark an order as recurring in the tablet, and require the player to keep the tablet in their inventory.  It's not perfect, but it's not bad either.

One thing I do need to make sure is completely robust is delivery drone tracking, i.e. making sure that a delivery isn't started until the previous delivery finished and the drones despawned properly.  Don't really want the situation where hundreds of delivery drones are piling up...
